### PR TITLE
Pantheon Write Problem: Update code with custom path for CONFIG & CACHE file

### PR DESCRIPTION
**Pull request by spinxdigital**
Pantheon has protected path and this plugin can't create file on protected path.
So we did path variable as custom global constant so it can change from wp-config.

With that it will help to put cache, config file to custom location like inside upload folder.

### CODE REVIEW

**Comment by spinxdigital** [comment]
@crystinutzaa  Can you please merge my commit? 
Also not sure why it's failed on last one.

File: uninstall.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 defined( 'WP_UNINSTALL_PLUGIN' ) || die( 'Cheatin&#8217; uh?' );
 
 if ( ! defined( 'WP_ROCKET_CACHE_ROOT_PATH' ) ) {
-	define( 'WP_ROCKET_CACHE_ROOT_PATH', WP_CONTENT_DIR . '/cache/' );
+	define( 'WP_ROCKET_CACHE_ROOT_PATH', WP_ROCKET_CACHE_ROOT_PATH );
 }
```
Comment (Tabrisrp) [inline uninstall.php]: that doesn't really make sense: if the constant is not defined, you can't assign a value that doesn't exist

Comment (spinxdigital) [reply uninstall.php]: Fixed constant code issue.

**Comment by spinxdigital** [comment]
Thank you for detailed review. @Tabrisrp 

I have fixed code constant error and revert back code about advanced-cache.php location.

Also found much batter solution with change full path to relative path which user can chick-in into source controller and should work for all dev, staging and live environment. 
But I am sure you know much batter test cases that if that will broke anything else or not.

File: inc/functions/files.php [hunk 2]
```diff
@@ -21,8 +21,8 @@ function get_rocket_advanced_cache_file() {
 
 	// Include the Mobile Detect class if we have to create a different caching file for mobile.
 	if ( is_rocket_generate_caching_mobile_files() ) {
-		$buffer .= "if ( file_exists( '" . WP_ROCKET_VENDORS_PATH . "classes/class-rocket-mobile-detect.php' ) && ! class_exists( 'Rocket_Mobile_Detect' ) ) {\n";
-		$buffer .= "\tinclude_once '" . WP_ROCKET_VENDORS_PATH . "classes/class-rocket-mobile-detect.php';\n";
+		$buffer .= "if ( file_exists( WP_CONTENT_DIR . '/plugins/wp-rocket/inc/vendors/classes/class-rocket-mobile-detect.php' ) && ! class_exists( 'Rocket_Mobile_Detect' ) ) {\n";
+		$buffer .= "\tinclude_once WP_CONTENT_DIR . '/plugins/wp-rocket/inc/vendors/classes/class-rocket-mobile-detect.php';\n";
```
Comment (Tabrisrp) [inline inc/functions/files.php]: What's the reasoning behind changing those paths? The constant `WP_ROCKET_VENDORS_PATH` is perfectly fine here

File: inc/functions/files.php [hunk 3]
```diff
@@ -48,19 +48,21 @@ function get_rocket_advanced_cache_file() {
...
 
+	$cache_dir_path = defined( \'WP_ROCKET_CACHE_ROOT_PATH\' ) ? WP_ROCKET_CACHE_ROOT_PATH : WP_CONTENT_DIR . \'/cache/wp-rocket/\';
 	( new \WP_Rocket\Buffer\Cache(
 		new \WP_Rocket\Buffer\Tests(
 			$rocket_config_class
 		),
 		$rocket_config_class,
 		[
-			\'cache_dir_path\' => \'' . WP_ROCKET_CACHE_PATH . '\',
+			\'cache_dir_path\' => $cache_dir_path,
```
Comment (Tabrisrp) [inline inc/functions/files.php]: These changes don't seem necessary to me either

File: inc/process-autoloader.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 spl_autoload_register(
 	function( $class ) {
-		$rocket_path    = WP_ROCKET_PATH;
+		$rocket_path    = WP_CONTENT_DIR . '\plugins\wp-rocket/';
```
Comment (Tabrisrp) [inline inc/process-autoloader.php]: same here

**Comment by spinxdigital** [comment]
@Tabrisrp 

I thought I can make path relative compare to absolute server path.
Like this path see on that file. `/srv/bindings/7580cf3199e348fca7cf1120d0c481a8/code/web/app/plugins/wp-rocket/` 
But I though I like previous version idea where path was like this. So it might use same file to different to dev, test and live server. But I think user can activate plugin on different environment and plugin will create that file.  
```
$rocket_cache_path = WP_CONTENT_DIR . '/uploads/cache/wp-rocket/';
$rocket_config_path = WP_CONTENT_DIR . '/uploads/wp-rocket-config/';
```

**Pull request merged**